utils/channel_intro_messages.jsx

```diff
@@ -387,3 +375,35 @@ function createSetHeaderButton(channel) {
         </ToggleModalButton>
     );
 }
+
+function showManagementOption(channel) {
+    const isChannelAdmin = ChannelStore.isChannelAdminForCurrentChannel();
+    const isTeamAdmin = TeamStore.isTeamAdminForCurrentTeam();
+    const isSystemAdmin = UserStore.isSystemAdminForCurrentUser();
+
+    if (!showManagementOptions(channel, isChannelAdmin, isTeamAdmin, isSystemAdmin)) {
+        return false;
+    }
+
+    return true;
+}
+
+function showComponent(permission) {
```

Comment:
The name `showComponent` is a little unclear to me, is there a more descriptive/accurate name that would make it's use obvious?

Reply:
Agree. I changed it to `isCurrentUserPermitted` and did some code refactor.